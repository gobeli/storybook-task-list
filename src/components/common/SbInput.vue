<template>
  <input
    class="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
    :class="{ 'border-4 border-red-300': $showBorders }"
    :name="name" :type="type" :placeholder="placeholder" :readonly="readonly" :value="value"
    @input="handleInput"
  >
</template>

<script>
export default {
  name: 'SbInput',
  methods: {
    handleInput(event) {
      this.$emit('input', event.target.value)
    },
  },
  props: {
    name: String,
    type: {
      type: String,
      default: 'text'
    },
    value: String,
    placeholder: String,
    readonly: Boolean
  }
}
</script>
