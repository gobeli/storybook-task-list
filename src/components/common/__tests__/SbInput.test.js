import { shallowMount } from '@vue/test-utils'

import SbInput from '../SbInput.vue'

describe('SbInput', () => {
  it('should render an input', () => {
    // given
    const vm = shallowMount(SbInput)

    // then
    expect(vm.element).toMatchSnapshot()
  })

  it('should be able to set value', () => {
    // given
    const vm = shallowMount(SbInput)
    const value = 'test';

    // when
    vm.setValue(value)

    // then
    expect(vm.element.value).toEqual(value)
  })
})