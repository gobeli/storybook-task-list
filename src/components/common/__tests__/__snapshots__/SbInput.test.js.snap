// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SbInput should render an input 1`] = `
<input
  class="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
  type="text"
/>
`;
